blocks/v2-article-search/v2-article-search.js

```diff
@@ -244,37 +244,47 @@ const initializeSearchHandlers = (searchContainer) => {
   }
 };
 
+const decorateArticleSearch = async (block, section, main) => {
```

Comment:
The other day, I reviewed how to resolve the bug and believe we can refactor and simplify this block of code. 
Rather than iterating through all mutations, we should directly check for the presence of `articleHero`.
Additionally, we could optimise performance by restricting the observer to monitor only `childList` changes on `main` with `subtree: false`. You can implement an early return when the `v2-magazine` class is not present, which would prevent unnecessary setup.
This is only an example, that would have to be adapted, but that might help.

```
export default async function decorate(block) {
  const main = block.closest('main');
  const section = block.closest(`.${containerClass}`);

  if (!document.body.classList.contains('v2-magazine')) {
    decorateArticleSearch(block, section, main);
    return;
  }

  const checkAndDecorate = () => {
    const articleHero = main.querySelector('.v2-magazine-article-hero__container');
    if (articleHero) {
      decorateArticleSearch(block, section, main);
      observer?.disconnect(); // Disconnect observer if it's defined
      return true;
    }
    return false;
  };

  const observer = new MutationObserver(() => checkAndDecorate());

  if (!checkAndDecorate()) {
    observer.observe(main, {
      childList: true,
      subtree: false,
    });
  }
}
```

Reply:
the logic has been moved to common.js to be like a sub-nav block